packages/block-editor/src/hooks/font-size.js

```diff
@@ -171,31 +172,30 @@ const withFontSizeInlineStyles = createHigherOrderComponent(
 			wrapperProps,
 		} = props;
 
-		// Return early if the block doesn't allow modify the font-size,
-		// already has a inline font-size,
-		// or doesn't have a class to extract font-size from.
+		// Only add inline styles if the block supports font sizes, doesn't
+		// already have an inline font size, and does have a class to extract
+		// the font size from.
 		if (
-			! hasBlockSupport( blockName, FONT_SIZE_SUPPORT_KEY ) ||
-			style?.typography?.fontSize ||
-			! fontSize
+			hasBlockSupport( blockName, FONT_SIZE_SUPPORT_KEY ) &&
+			fontSize &&
+			! style?.typography?.fontSize
 		) {
-			return <BlockListBlock { ...props } />;
+			const fontSizeValue = getFontSize(
+				fontSizes,
+				fontSize,
+				style?.typography?.fontSize
+			).size;
+
+			props.wrapperProps = {
```

Comment:
I don't like mutating arguments personally. Ideally, we'd have a lint rule for that as it can potentially create weird issues. (Not certain it's the case here though)

Reply:
It's spread a few lines below, so it should be ok

Reply:
It should be spread before making the change. The problem here is that you're changing the value of an object defined elsewhere (the caller) which means potentially unexpected side effects there.

Reply:
This breaks throwing:
`Uncaught TypeError: can't define property "wrapperProps": Object is not extensible`

Reply:
Fixed by @Copons in #23717